.ci/.lintr.R

```diff
@@ -71,45 +71,39 @@ linters = c(dt_linters, all_linters(
 ))
 rm(dt_linters)
 
-# TODO(lintr#2172): Glob with lintr itself.
-exclusions = c(local({
-  exclusion_for_dir <- function(dir, exclusions) {
-    files = file.path("..", list.files(dir, pattern = "\\.(R|Rmd|Rraw)$", full.names=TRUE))
-    stats::setNames(rep(list(exclusions), length(files)), files)
-  }
-  c(
-    exclusion_for_dir("tests", list(
-      quotes_linter = Inf,
-      # TODO(michaelchirico): Enforce these and re-activate them one-by-one.
-      implicit_integer_linter = Inf,
-      infix_spaces_linter = Inf,
-      undesirable_function_linter = Inf
-    )),
-    exclusion_for_dir(c("vignettes", "vignettes/fr", "vignettes/ru"), list(
-      # assignment_linter = Inf,
-      implicit_integer_linter = Inf,
-      quotes_linter = Inf,
-      sample_int_linter = Inf
-      # strings_as_factors_linter = Inf
-      # system_time_linter = Inf
-    )),
-    exclusion_for_dir("inst/tests", list(
-      library_call_linter = Inf,
-      numeric_leading_zero_linter = Inf,
-      undesirable_operator_linter = Inf, # For ':::', possibly we could be more careful to only exclude ':::'.
-      # TODO(michaelchirico): Enforce these and re-activate them one-by-one.
-      comparison_negation_linter = Inf,
-      condition_call_linter = Inf,
-      duplicate_argument_linter = Inf,
-      equals_na_linter = Inf,
-      missing_argument_linter = Inf,
-      paste_linter = Inf,
-      rep_len_linter = Inf,
-      sample_int_linter = Inf,
-      seq_linter = Inf,
-      unnecessary_lambda_linter = Inf
-    ))
+exclusions = list(
+  `../tests` = list(
+    quotes_linter = Inf,
+    # TODO(michaelchirico): Enforce these and re-activate them one-by-one.
+    implicit_integer_linter = Inf,
+    infix_spaces_linter = Inf,
+    undesirable_function_linter = Inf
+  ),
+  `../vignettes*` = list(
```

Comment:
does the star work for sub-directories? fr etc?

Reply:
well it's ignoring the whole directory, so yes subdirectories are included.

I can't see why we'd want different configs for different languages, but anyway this FR could eventually be relevant:

https://github.com/r-lib/lintr/issues/1976